cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/PreferencesHttpHandler.java

```diff
@@ -74,6 +85,150 @@
     this.nsStore = nsStore;
   }
 
+  /**
+   * Returns user repository information
+   */
+  @Path("git/repo/info")
+  @GET
+  public void getRepoInfo(HttpRequest request, HttpResponder responder) throws Exception {
+    //check if repo info exists
+    if (userRepo.toString().equals("{}")) {
+      responder.sendJson(HttpResponseStatus.BAD_REQUEST, GSON.toJson("Please add repo information first."));
+    } else {
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(userRepo));
+    }
+  }
+
+  @Path("git/repo/add")
+  @PUT
+  public void addRepoInfo(FullHttpRequest request, HttpResponder responder) throws Exception {
+    //first check if repository info has already been saved
+    if (!userRepo.toString().equals("{}")) {
+      responder.sendJson(HttpResponseStatus.BAD_REQUEST, GSON.toJson("Repo Information is already saved"));
+    } else {
+      JsonObject jObj = DECODE_GSON.fromJson(request.content().toString(StandardCharsets.UTF_8), JsonObject.class);
```

Comment:
Try to create a class for your repo creation request instead of parse it to plain json object. 

Please refer : https://github.com/cdapio/cdap/blob/develop/cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ConnectionHandler.java#L153